keras_cv/layers/vit_patch_encoding.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+from keras_cv.layers.vit_class_tokenizing import ClassTokenizing
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class PatchEncoding(layers.Layer):
```

Comment:
In the HF modularization we have the opportunity to optionally interpolate the position embedding to fine-tune VIT at higher resolution:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/vit/modeling_tf_vit.py#L742-L744

Reply:
Is `PatchEncoding` an expected name? Or "PosEmbeddings" etc..

Reply:
The original paper refers to them as patch embeddings. Might be better to rename the layer `PatchEmbedding` or `PathEmbeddings`?

> We refer to the output of this projection as the patch embeddings. Similar to BERT’s [class] token, we prepend a learnable embedding to the sequence [...]. Position embeddings are added to the patch embeddings to retain positional information.

Reply:
Also, good point on the interpolation flag! I'll take a look and try to add it in :)

Reply:
Added the `interpolate` flag! The patch sizes would be passed by the configuration of the model in the classfile. The rest (interpolation width/height and whether they want to interpolate) would be left to the user.

Updated the example Google Colab notebook with it too

Reply:
Added support for interpolation! Can you take a look?